lib/experimental/OneChip/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { Icon, IconType } from "@/components/Utilities/Icon"
+import {
+  Avatar,
+  AvatarVariant,
+} from "@/experimental/Information/Avatars/Avatar"
+import { CrossedCircle } from "@/icons/app"
+import { cn, focusRing } from "@/lib/utils"
+import { cva, type VariantProps } from "cva"
+
+export const chipVariants = cva({
+  base: "flex items-center gap-1 rounded-full border border-solid border-f1-border px-2 py-0.5 font-medium",
+  variants: {
+    variant: {
+      default: "",
+      selected:
+        "border-f1-border-selected bg-f1-background-selected-secondary text-f1-foreground-selected",
+    },
+  },
+  defaultVariants: {
+    variant: "default",
+  },
+})
+
+interface ChipProps extends VariantProps<typeof chipVariants> {
+  /**
+   * The label of the chip
+   */
+  label: string
+
+  /**
+   * If defined, the close icon will be displayed and the chip will be clickable
+   */
+  onClose?: () => void
+
+  /**
+   * If defined, an avatar will be displayed in the chip
+   */
+  avatar?: AvatarVariant
+
+  /**
+   * If defined, an icon will be displayed in the chip
+   */
+  icon?: IconType
+}
```

Comment:
If icon and avatar can not be defined at same time, this type ensures that
```suggestion
interface ChipProps extends VariantProps<typeof chipVariants> {
  /** The label of the chip */
  label: string
  
  /** If defined, the close icon will be displayed and the chip will be clickable */
  onClose?: () => void
  
} & (
  | { /** If defined, an avatar will be displayed in the chip */
      avatar: AvatarVariant
      icon?: never
    }
  | { /** If defined, an icon will be displayed in the chip */
      icon: IconType
      avatar?: never
    }
  | { 
      avatar?: never
      icon?: never
    }
)
```

Reply:
Ah wow, this is magical, thanks! 🫶